src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -395,6 +395,13 @@ private async Task ProduceEndAwaited()
         await _bodyOutput.FlushAsync(default);
     }
 
+    // Response trailers, reset, and GOAWAY are only on HTTP/2+ and require IIS support
+    // that is only available on Win 11/Server 2022 or later.
+    private static readonly bool isCompatibleOsVersion = OperatingSystem.IsWindowsVersionAtLeast(10, 0, 20348, 0);
+    protected bool AdvancedHttp2FeaturesSupported => isCompatibleOsVersion &&
```

Comment:
```suggestion
    private static readonly bool AdvancedHttp2CompatibleOsVersion = OperatingSystem.IsWindowsVersionAtLeast(10, 0, 20348, 0);
    protected bool AdvancedHttp2FeaturesSupported => AdvancedHttp2CompatibleOsVersion &&
```

Reply:
This name reads like it's a version instead of a bool. How about OsSupportsAdvancedHttp2?

Reply:
`SupportsAdvancedHttp2Features`?

Reply:
I want to emphasize OS on this one since it's one specific part of the requirements for `AdvancedHttp2FeaturesSupported`.